ERCS/erc-7579.md

```diff
@@ -0,0 +1,351 @@
+
+### Specifications
+
+#### Multiple execution functions
+
+The ERC-4337 validation phase validates the `UserOperation`. Modular validation requires the validation module to know the specific function being validated, especially for session key based Validators. It needs to know:
+
+1. The function called by Entrypoint on the account.
+2. The target address if it is an execution function.
+3. Whether it is a `call` or `delegatecall`.
+4. Whether it is a single or batched transaction.
+5. The function signature used in the interaction with the external contract.
+
+For a flourishing module ecosystem, compatibility across accounts is crucial. However, if smart accounts implement custom execute functions with different parameters and calldata offsets, it becomes much harder to build reusable modules across accounts.
+
+#### Differentiating module types
+
+Not differentiating between module types could present a security issue when enforcing authorization control. For example, if a smart account treats validators and executors as the same type of module, it could allow a validator to execute arbitrary transactions on behalf of the smart account.
+
+#### Dependence on ERC-4337
+
+This standard has a strict dependency on ERC-4337 for the validation flow. However, it is likely that smart account builders will want to build modular accounts in the future that do not use ERC-4337 but, for example, a native account abstraction implementation on a rollup. Once this starts to happen, the proposed upgrade path for this standard is to move the ERC-4337 dependency into an extension (ie a separate ERC) and to make it optional for smart accounts to implement. If it is required to standardize the validation flow for different account abstraction implementations, then these requirements could also be moved into separate extensions.
+
+The reason this is not done from the start is that currently, the only modular accounts that are being built are using ERC-4337. Therefore, it makes sense to standardize the interfaces for these accounts first and to move the ERC-4337 dependency into an extension once there is a need for it. This is to maximize learnings about how modular accounts would look like when built on different account abstraction implementations.
+
+## Backwards Compatibility
+
+### Already deployed smart accounts
+
+Smart accounts that have already been deployed will most likely be able to implement this standard. If they are deployed as proxies, it is possible to upgrade to a new account implementation that is compliant with this standard. If they are deployed as non-upgradeable contracts, it might still be possible to become compliant, for example by adding a compliant adapter as a fallback handler, if this is supported.
+
+## Reference Implementation
+
+A full interface of a smart account can be found in [`IMSA.sol`](../assets/eip-7579/IMSA.sol). A reference implementation will follow soon.
```

Comment:
```suggestion
A full interface of a smart account can be found in [`IMSA.sol`](../assets/eip-7579/IMSA.sol).
```

You should write your proposals as if they are already finished. It wouldn't make sense to have this comment in the Final version of the proposal.

If you like, you can use an HTML-style comment (`<!-- TODO: add reference implementation -->`) and the linter will remind you to remove it.